lib/client/ipam_block_reader_writer.go

```diff
@@ -55,7 +55,9 @@ func (rw blockReaderWriter) getAffineBlocks(host string, ver ipVersion, pool *cn
 	ids := []cnet.IPNet{}
 	for _, o := range datastoreObjs {
 		k := o.Key.(model.BlockAffinityKey)
-		ids = append(ids, k.CIDR)
+		if reflect.DeepEqual(k.CIDR.IPNet, pool.IPNet) {
```

Comment:
This is wrong, we should be checking the block is in the pool, not equal to.  Test also needs correcting, as discussed.